internal/pkg/composable/controller.go

```diff
@@ -208,10 +221,22 @@ func (c *controller) Run(ctx context.Context) error {
 			}
 		}
 
-		select {
-		case c.ch <- vars:
-		case <-ctx.Done():
-			// coordinator is handling cancellation it won't drain the channel
+	UPDATEVARS:
+		for {
+			select {
+			case c.ch <- vars:
+				break UPDATEVARS
+			case <-ctx.Done():
+				// coordinator is handling cancellation it won't drain the channel
+			default:
+				// c.ch is size of 1, nothing is reading and there's already a signal
+				select {
+				case <-c.ch:
```

Comment:
i agree with refactoring option as a followup. i'd rather have notifications in a form of .net events where producer does not need to consider readers. 
as we talked about, i wanted this PR to be as least intrusive as possible so we can spend proper time with refactor without time pressure from production issues.

Reply:
Create an issue for the refactor so we don't forget to do it please :)

Reply:
What is the possibility that this results in a vars set being lost? Confused on why this is really needed? There should always be a read and if there is not a reader the controller should really block to ensure that all vars are read.

Reply:
idea behind is that when controller is handling something else this will become blocking,
when this is blocked no new vars are being processed and stored in mappings because we have a loop stuck in here. so this helps controller to get most recent mappings based on actual events

Reply:
I just worry that if we empty the channel like it is here, then the reader of the channel starts reading again the set of variables will be removed and then it will not get the variables from the channel.

Reply:
should not happen when we drain we dont cancel a loop and will push vars again in next cycle